x/stakeibc/keeper/community_pool_test.go

```diff
@@ -166,3 +175,137 @@ func (s *KeeperTestSuite) TestQueryCommunityPoolBalance_FailedQuerySubmission()
+	s.FundAccount(communityPoolHoldingAddress, sdk.NewCoin(StAtom, initialDummyTokens)) // dummy token
+
+	return LiquidStakeCommunityPoolTokensTestCase{
+		hostZone:            hostZone,
+		initialNativeTokens: initialNativeTokens,
+		initialDummyTokens:  initialDummyTokens,
+	}
+}
+
+func (s *KeeperTestSuite) TestLiquidStakeCommunityPoolTokens_Success() {
+	tc := s.SetupLiquidStakeCommunityPoolTokens()
+
+	transferPortId := transfertypes.PortID
+	transferChannelId := ibctesting.FirstChannelID
+	communityPoolHoldingAddress := sdk.MustAccAddressFromBech32(tc.hostZone.CommunityPoolStakeHoldingAddress)
+
+	// Call liquid stake which should convert the whole native tokens amount to stTokens and transfer it
+	err := s.App.StakeibcKeeper.LiquidStakeCommunityPoolTokens(s.Ctx, tc.hostZone)
+	s.Require().NoError(err, "no error expected during liquid stake")
+
+	// Confirm there are no longer native tokens in the holding address
+	ibcAtomBalance := s.App.BankKeeper.GetBalance(s.Ctx, communityPoolHoldingAddress, IbcAtom)
+	s.Require().Zero(ibcAtomBalance.Amount.Int64(), "balance of holding address should be zero")
+
+	// Confirm the dummy tokens are still present
+	dummyTokenBalance1 := s.App.BankKeeper.GetBalance(s.Ctx, communityPoolHoldingAddress, Atom)
+	dummyTokenBalance2 := s.App.BankKeeper.GetBalance(s.Ctx, communityPoolHoldingAddress, StAtom)
+	s.Require().Equal(tc.initialDummyTokens, dummyTokenBalance2.Amount, "dummy token 2 was not touched")
+	s.Require().Equal(tc.initialDummyTokens, dummyTokenBalance1.Amount, "dummy token 1 was not touched")
+
+	// Confirm the stTokens have been escrowed as a result of the transfer
+	escrowAddress := transfertypes.GetEscrowAddress(transferPortId, transferChannelId)
+	stTokenEscrowBalance := s.App.BankKeeper.GetBalance(s.Ctx, escrowAddress, StAtom)
+	s.Require().Equal(tc.initialNativeTokens.Int64(), stTokenEscrowBalance.Amount.Int64(), "st token escrow balance")
+
+	// Check that if we run the liquid stake function again, nothing should get transferred
+	startSequence, found := s.App.IBCKeeper.ChannelKeeper.GetNextSequenceSend(s.Ctx, transferPortId, transferChannelId)
+	s.Require().True(found, "sequence number not found before liquid stake")
+
+	err = s.App.StakeibcKeeper.LiquidStakeCommunityPoolTokens(s.Ctx, tc.hostZone)
+	s.Require().NoError(err, "no error expected during second liquid stake")
+
+	endSequence, found := s.App.IBCKeeper.ChannelKeeper.GetNextSequenceSend(s.Ctx, transferPortId, transferChannelId)
+	s.Require().True(found, "sequence number not found after after liquid stake")
+
+	s.Require().Equal(startSequence, endSequence, "no transfer should have been initiated")
+}
+
+// Test liquid stake with an invalid stake holding address
+func (s *KeeperTestSuite) TestLiquidStakeCommunityPoolTokens_Failure_InvalidAddress() {
+	tc := s.SetupLiquidStakeCommunityPoolTokens()
+
+	invalidHostZone := tc.hostZone
+	invalidHostZone.CommunityPoolStakeHoldingAddress = "invalid"
```

Comment:
do you need to call s.App.StakeibcKeeper.SetHostZone(s.Ctx, invalidHostZone) after this modification?

Reply:
good question! we actually don't since the host zone is passed as a param instead of being looked up in this function